roles/add-networks/tasks/main.yml

```diff
@@ -0,0 +1,48 @@
+---
+- name: get node and master addresses
+  shell: "openstack coe cluster show {{ cluster_name }} -c master_addresses -c node_addresses -f json"
+  register: result_addresses
+
+- name: get list of servers
+  shell: "openstack server list -c ID -c Networks -f json"
+  register: result_servers
+
+- name: convert results to json
+  set_fact:
+    addresses: "{{ result_addresses.stdout | from_json }}"
+    servers: "{{ result_servers.stdout | from_json }}"
+
+- name: add masters
+  add_host:
+    hostname: "{{ item[1].ID }}"
+    ansible_host: "{{ item[0] }}"
+    ansible_user: "{{ cluster_user }}"
+    ansible_network_os: "{{ cluster_os }}"
+    networks: "{{ item[1].Networks.split('; ') | join('\n') | regex_replace('=', ': ') | from_yaml }}"
+    groups:
+    - masters
+    - cluster
+  when: item[1].Networks is search(item[0])
+  loop: "{{ query('nested', addresses.master_addresses, servers) }}"
+
+- name: add workers
+  add_host:
+    hostname: "{{ item[1].ID }}"
+    ansible_host: "{{ item[0] }}"
+    ansible_user: "{{ cluster_user }}"
+    networks: "{{ item[1].Networks.split('; ') | join('\n') | regex_replace('=', ': ') | from_yaml }}"
+    groups:
+    - workers
+    - cluster
+  when: item[1].Networks is search(item[0])
+  loop: "{{ query('nested', addresses.node_addresses, servers) }}"
+
+- debug:
+    var: hostvars
+
+- name: attach interfaces
```

Comment:
`name: Attach COE host interfaces to IB network`?

Reply:
Sounds better!